File: backend/GestaoClientesPedidos.Core/Entities/Pedido.cs
using GestaoClientesPedidos.Core.Enums;

namespace GestaoClientesPedidos.Core.Entities;

public class Pedido : BaseEntity
{
    public int ClienteId { get; set; }
    public Cliente Cliente { get; set; } = null!;

    public DateTime DataPedido { get; set; } = DateTime.UtcNow;
    public StatusPedido Status { get; set; } = StatusPedido.Pendente;
    public decimal ValorTotal { get; set; }
    public decimal Desconto { get; set; }
    public string? Observacoes { get; set; }

    // Navigation property
    public ICollection<PedidoItem> Itens { get; set; } = new List<PedidoItem>();

    public void CalcularTotal()
    {
        ValorTotal = Itens.Sum(i => i.Subtotal) - Desconto;
        if (ValorTotal < 0) ValorTotal = 0;
    }
}

File: frontend/src/types/pedido.ts
export type StatusPedido = 'Pendente' | 'Processando' | 'Enviado' | 'Concluido' | 'Cancelado';

export interface PedidoItem {
  id: number;
  produtoId: number;
  produtoNome: string;
  produtoSku: string;
  quantidade: number;
  precoUnitario: number;
  subtotal: number;
}

export interface Pedido {
  id: number;
  clienteId: number;
  clienteNome: string;
  clienteEmail: string;
  clienteDocumento: string;
  dataPedido: string;
  status: StatusPedido;
  statusDescricao: string;
  valorTotal: number;
  desconto: number;
  observacoes?: string;
  itens: PedidoItem[];
}

export interface CreatePedidoItemRequest {
  produtoId: number;
  quantidade: number;
}

export interface CreatePedidoRequest {
  clienteId: number;
  desconto: number;
  observacoes?: string;
  itens: CreatePedidoItemRequest[];
}

export interface UpdatePedidoStatusRequest {
  status: StatusPedido;
}
